### PR TITLE
update pluginData name&version, update listener name

### PR DESCRIPTION
**Pull request by haomingzhang**

### CODE REVIEW

File: common/data_structs.go [hunk 1]
```diff
@@ -14,9 +14,8 @@
 package common
 
 type Attribute struct {
-	Name  string `json:"name,omitempty"`
-	Value string `json:"value,omitempty"`
-	Kind  string `json:"kind,omitempty"`
+	Name  string `json:"name"`
+	Value string `json:"value"`
```
Comment (haomingzhang) [inline common/data_structs.go]: required by Adi: do not omit empty fields in JSON

File: common/pluginData.go [hunk 2]
```diff
@@ -17,9 +17,9 @@ package common
 import "github.com/apid/apid-core"
 
 var PluginData = apid.PluginData{
-	Name:    "apidVerifyAPIKey",
-	Version: "0.1.0",
+	Name:    "apidApiMetadata",
```
Comment (chebrolus) [inline common/pluginData.go]: Would it be beneficial to use global constants (or is it too expensive to have global constants)?

File: listener.go [hunk 3]
```diff
@@ -34,7 +34,7 @@ func (h *apigeeSyncHandler) initListener(services apid.Services) {
 }
 
 func (h *apigeeSyncHandler) String() string {
-	return "verifyAPIKey"
+	return "apiMetadata"
```
Comment (chebrolus) [inline listener.go]: Sounds like this string literal also repeated, what is best practice in `Golang` world for such usage?

File: common/pluginData.go (hunk 2 above)
Comment (haomingzhang) [reply common/pluginData.go]: Yes, we can have constants.

File: listener.go (hunk 3 above)
Comment (haomingzhang) [reply listener.go]: This `String()` implements an interface for apid-core event handler.
It provides logging info for apid-core to log system events life-cycle.
By convention it's plugin name without "apid" prefix.

**Pull request merged**